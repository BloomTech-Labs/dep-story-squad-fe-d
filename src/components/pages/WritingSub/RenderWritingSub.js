import React from 'react';
import { Row } from 'antd';
import { connect } from 'react-redux';

import { Header } from '../../common';
import { UploadDocs } from '../../common/';
import { postNewWritingSub } from '../../../api/index';
import { InstructionsModal } from '../../common/index';
import { modalInstructions } from '../../../utils/helpers';
import { tasks } from '../../../state/actions';

export const RenderWritingSub = props => {
  return (
    <>
      <Header title="PENCILS READY?" />
      <InstructionsModal
        instructions={modalInstructions.writingSub}
        style={{ fontSize: '1.5rem' }}
      />
      <div className="writing-sub-container">
        <Row className="main-row">
          <p>{props.tasks.writingPrompt}</p>
        </Row>
        <div className="upload">
          <h1>Upload</h1>
          <UploadDocs
            submitButtonClassname="orange-submit-button"
            uploadButtonText="Choose files from your device"
            uploadButtonClassname="uploadButton"
            fileName="pages"
            apiAxios={postNewWritingSub}
            submissionId={props.tasks.id}
            storyId={props.tasks.story_id}
            setSubmitted={props.setHasWritten}
            maxLength={5}
          />
        </div>
      </div>
    </>
  );
};

export default connect(
  state => ({
    tasks: state.tasks,
  }),
  {
    setHasWritten: tasks.setHasWritten,
  }
)(RenderWritingSub);
