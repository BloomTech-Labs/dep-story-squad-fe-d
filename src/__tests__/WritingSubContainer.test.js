import * as React from 'react';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import WritingSubContainer from '../components/pages/WritingSub/WritingSubContainer';
import LoadingComponent from '../components/common/ParentLoadingComponent';
import RenderWritingSub from '../components/pages/WritingSub/RenderWritingSub';

jest.mock('@okta/okta-react', () => ({
  useOktaAuth: () => {
    return {
      authState: {
        isAuthenticated: true,
      },
      authService: {},
    };
  },
}));

describe('<WritingSubContainer />', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore([]);
  const store = mockStore();

  describe('Render WritingSubContainer', () => {
    let shallowWrapper = ShallowWrapper;
    beforeEach(() => {
      shallowWrapper = shallow(
        <Router>
          <Provider store={store}>
            <WritingSubContainer />
          </Provider>
        </Router>
      );
    });

    it('Find Loading Component', () => {
      expect(shallowWrapper.find(LoadingComponent));
    });
    it('Find RenderWritingSub', () => {
      expect(shallowWrapper.find(RenderWritingSub).length).toBe(0);
    });
  });
});
