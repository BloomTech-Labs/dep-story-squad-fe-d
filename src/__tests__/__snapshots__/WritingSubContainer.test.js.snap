// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileModalContainer /> Render <ProfileModalContainer /> Find WritingSubContainer 1`] = `ShallowWrapper {}`;

exports[`<WritingSubContainer /> Render WritingSubContainer Find ParentDashboard 1`] = `ShallowWrapper {}`;

exports[`<WritingSubContainer /> Render WritingSubContainer Find WritingSubContainer 1`] = `ShallowWrapper {}`;
